src/streamlink/logger.py

```diff
@@ -1,61 +1,212 @@
         return LoggerModule(self, module)
 
     def set_level(self, level):
-        try:
-            index = Logger.Levels.index(level)
-        except ValueError:
-            return
-
-        self.level = index
+        self.root.setLevel(level)
 
     def set_output(self, output):
-        self.output = output
+        self.handler.stream = output
 
     def msg(self, module, level, msg, *args, **kwargs):
-        if self.level < level or level > len(Logger.Levels):
-            return
-
-        msg = msg.format(*args, **kwargs)
-
-        with self.lock:
-            self.output.write(Logger.Format.format(module=module,
-                                                   level=Logger.Levels[level],
-                                                   msg=msg))
-            if hasattr(self.output, "flush"):
-                self.output.flush()
+        log = self.get_logger(module)
+        log.log(level, msg, *args)
 
 
 class LoggerModule(object):
     def __init__(self, manager, module):
+        warnings.warn("LoggerModule class has been deprecated, use the standard logging module",
+                      category=DeprecationWarning, stacklevel=2)
         self.manager = manager
         self.module = module
 
     def error(self, msg, *args, **kwargs):
-        self.manager.msg(self.module, 1, msg, *args, **kwargs)
+        self.manager.msg(self.module, ERROR, msg, *args, **kwargs)
 
     def warning(self, msg, *args, **kwargs):
-        self.manager.msg(self.module, 2, msg, *args, **kwargs)
+        self.manager.msg(self.module, WARN, msg, *args, **kwargs)
 
     def info(self, msg, *args, **kwargs):
-        self.manager.msg(self.module, 3, msg, *args, **kwargs)
+        self.manager.msg(self.module, INFO, msg, *args, **kwargs)
 
     def debug(self, msg, *args, **kwargs):
-        self.manager.msg(self.module, 4, msg, *args, **kwargs)
+        self.manager.msg(self.module, DEBUG, msg, *args, **kwargs)
+
+
+BASIC_FORMAT = "[{name}][{levelname}] {message}"
+FORMAT_STYLE = "{"
+REMOVE_BASE = ["streamlink", "streamlink_cli"]
+
+
+def basicConfig(**kwargs):
+    with _config_lock:
+        filename = kwargs.get("filename")
+        if filename:
+            mode = kwargs.get("filemode", 'a')
+            handler = logging.FileHandler(filename, mode)
+        else:
+            stream = kwargs.get("stream")
+            handler = logging.StreamHandler(stream)
+        fs = kwargs.get("format", BASIC_FORMAT)
+        style = kwargs.get("style", FORMAT_STYLE)
+        dfs = kwargs.get("datefmt", None)
+        remove_base = kwargs.get("remove_base", REMOVE_BASE)
+
+        formatter = StringFormatter(fs, dfs, style=style, remove_base=remove_base)
+        handler.setFormatter(formatter)
+
+        root.addHandler(handler)
+        level = kwargs.get("level")
+        if level is not None:
+            root.setLevel(level)
 
 
-__all__ = ["Logger"]
+__all__ = ["StreamlinkLogHandler", "Logger", "basicConfig", "root", "levels"]
```

Comment:
it's not used anywhere

`src/streamlink/logger.py:212:1: F822 undefined name 'StreamlinkLogHandler' in __all__`